sdk/dotnet/Pulumi.Automation/WorkspaceStack.cs

```diff
@@ -517,17 +517,34 @@ public Task RemoveConfigAsync(IEnumerable<string> keys, CancellationToken cancel
         /// <summary>
         /// Returns a list summarizing all previews and current results from Stack lifecycle operations (up/preview/refresh/destroy).
         /// </summary>
-        public async Task<ImmutableList<UpdateSummary>> GetHistoryAsync(CancellationToken cancellationToken = default)
+        /// <param name="limit">Limit the number of history entries to retrieve, defaults to all.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        public async Task<ImmutableList<UpdateSummary>> GetHistoryAsync(
+            int? limit = null,
+            CancellationToken cancellationToken = default)
         {
-            var result = await this.RunCommandAsync(new[] { "history", "--json", "--show-secrets" }, null, cancellationToken).ConfigureAwait(false);
+            var args = new List<string>()
+            {
+                "history",
+                "--json",
+                "--show-secrets",
+            };
+
+            if (limit.HasValue)
+            {
+                args.Add("--limit");
```

Comment:
Nice catch! I went back before we released this change and exposed the full set of pagination options via https://github.com/pulumi/pulumi/pull/6292

This should now be `--page-size` and `--page`

Reply:
missed that, should be good now!